### DISCUSSION TITLE
add SSAARenderPass

### DISCUSSION THREAD

**Discussion by J12My** [enable3d/enable3d#276] [Help / Questions]
Hello everyone,

I would like to add a SSAARenderPass but I have an issue with EffectComposer

<img width="1556" alt="Capture d’écran 2022-05-17 à 12 15 50" src="https://user-images.githubusercontent.com/70979321/168789051-8822fdce-26e0-4ebe-a845-77cd9c3fd84c.png">

I tried to follow the exemple Post-Processing
https://github.com/enable3d/enable3d-website/blob/master/src/examples/post-processing.html

but I really don't understand why it's not working

`
const { AmmoPhysics, PhysicsLoader, ExtendedObject3D, EffectComposer, RenderPass, ShaderPass } = ENABLE3D

const MainScene = () => {

  const scene = new THREE.Scene()
  scene.background = new THREE.Color(0x000000)

  const camera = new THREE.OrthographicCamera(document.getElementById("view3d").offsetWidth*-1 , document.getElementById("view3d").offsetWidth, document.getElementById("view3d").offsetHeight, document.getElementById("view3d").offsetHeight*-1, -200, 6000);
  camera.position.set(1000, 1000, 2000)

  const renderer = new THREE.WebGLRenderer()
  renderer.setSize(document.getElementById("view3d").offsetWidth, document.getElementById("view3d").offsetHeight)
  console.log(renderer)
  var composer = new EffectComposer(renderer)

  // setup effects
  var renderPass = new RenderPass(scene, camera)
  composer.addPass(renderPass)

  var effectSSAA = new THREE.SSAARenderPass(scene, camera)
  effectSSAA.renderToScreen = true
  composer.addPass(effectSSAA)

  document.getElementById("view3d").appendChild(renderer.domElement)

  const controls = new THREE.OrbitControls(camera, document.getElementById("view3d"))
  controls.autoRotate = false;

  btnTopView.addEventListener('click', function(){
    controls.reset()
    camera.position.set(0, 100, 0)
    camera.lookAt(new THREE.Vector3(0, 0, 0))
  })
  
  btnFreeView.addEventListener('click', function(){
    controls.reset();
    camera.position.set(1000, 1000, 2000);
    camera.lookAt(new THREE.Vector3(0, 0, 0));
  })

  scene.add(new THREE.HemisphereLight(0xffffbb, 0x080820, 1))
  scene.add(new THREE.AmbientLight(0x666666))
  const light = new THREE.DirectionalLight(0xdfebff, 1)
  light.position.set(2000, 2000, 2000)
  light.position.multiplyScalar(1.3)

  // add axis
  var axes = new THREE.AxesHelper(800);
  scene.add(axes);

  //add pipe
  var pipeGeometry = new THREE.CylinderGeometry(10, 10, 200, 200)
  var pipeMaterial = new THREE.MeshToonMaterial({color: 0xFFCC00})
  const pipe = new THREE.Mesh(pipeGeometry, pipeMaterial)
  pipe.position.set(0, 0, 0)
  pipe.castShadow = true;
  pipe.receiveShadow = true;
  scene.add(pipe)

  const clock = new THREE.Clock()

  const animate = () => {

    pipe.scale.x = ((pipeDiam.value - slider.value) / 2) / 10
    pipe.scale.z = ((pipeDiam.value - slider.value) / 2) / 10

    //renderer.render(scene, camera)
    composer.render();
    requestAnimationFrame(animate)
  }
  requestAnimationFrame(animate)
}
PhysicsLoader('./src/lib/ammo/kripken', () => MainScene())

console.log(`three.js version "${THREE.REVISION}"`)
`

In my HTML file I call the SSAARenderPass.js

`
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="./src/css/style.css">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" integrity="sha512-KfkfwYDsLkIlwQp6LFnl8zNdLGxu9YAA1QvwINks4PhcElQSvqcyVLLD9aMhXd13uQjoXtEKNosOWaZqXgel0g==" crossorigin="anonymous" referrerpolicy="no-referrer" />
    <link rel="preconnect" href="https://fonts.googleapis.com"> 
    <link rel="preconnect" href="https://fonts.googleapis.com"> 
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
    <link href="https://fonts.googleapis.com/css2?family=Roboto+Condensed:wght@400;700&display=swap" rel="stylesheet">
    
    <script src="./src/lib/three.min.js?ver=r130"></script>
    <script src="./src/lib/GLTFLoader.js"></script>
    <script src="./src/lib/enable3d/enable3d.ammoPhysics.0.25.1.min.js"></script>
    <script src="./src/lib/OrbitControls.js"></script>

    <script src="./src/lib/SSAA/SSAARenderPass.js"></script>
    
    <title>test</title>
</head>
`

What I'm doing wrong ?

Thanks in advance for your help

**Comment by yandeu** [comment]
Since you are not using the enable3d "framework", please use the threejs forum instead:
https://discourse.threejs.org

Also, read this:
https://threejs.org/docs/#manual/en/introduction/How-to-use-post-processing
